pkg/agent/cniserver/interface_configuration_linux_test.go

```diff
@@ -0,0 +1,857 @@
+			ovsHardwareOffloadEnabled: true,
+			podSriovVFDeviceID:        "sriovPodVF",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			defer mockSetupVethWithName(tc.setupVethErr, 1, 2)()
+			defer mockRenameInterface(tc.renameIntefaceErr)()
+			defer mockIPAMConfigureIface(tc.ipamConfigureIfaceErr)()
+			defer mockEthtoolTXHWCsumOff(tc.ethtoolEthTXHWCsumOffErr)()
+			testIfConfigurator := newTestIfConfigurator(tc.ovsHardwareOffloadEnabled, fakeNetlink, fakeSriovNet)
+			containerNS := createNS(t, false)
+			moveVFtoNS := false
+			if tc.sriovVFDeviceID != "" && tc.ovsHardwareOffloadEnabled {
+				fakeSriovNet.EXPECT().GetNetDevicesFromPci(tc.sriovVFDeviceID).Return(tc.vfNetdevices, nil).Times(1)
+				if len(tc.vfNetdevices) == 1 {
+					fakeSriovNet.EXPECT().GetUplinkRepresentor(tc.sriovVFDeviceID).Return(sriovUplinkName, nil).Times(1)
+					fakeSriovNet.EXPECT().GetVfIndexByPciAddress(tc.sriovVFDeviceID).Return(sriovVfIndex, nil).Times(1)
+					fakeSriovNet.EXPECT().GetVfRepresentor(sriovUplinkName, sriovVfIndex).Return(sriovVfRepresentor, nil).Times(1)
+					if tc.renameIntefaceErr == nil {
+						hostInterfaceLink := &netlink.Dummy{
+							LinkAttrs: netlink.LinkAttrs{Index: 2, MTU: mtu, HardwareAddr: hostVethMac, Name: hostIfaceName, Flags: net.FlagUp},
+						}
+						fakeNetlink.EXPECT().LinkByName(hostIfaceName).Return(hostInterfaceLink, nil).Times(1)
+						moveVFtoNS = true
+					}
+				}
+			}
+			if tc.podSriovVFDeviceID != "" {
+				fakeSriovNet.EXPECT().GetPfName(tc.podSriovVFDeviceID).Return(sriovPfName, nil).Times(1)
+				fakeSriovNet.EXPECT().GetVfid(tc.podSriovVFDeviceID, sriovPfName).Return(sriovVfIndex, nil).Times(1)
+				fakeSriovNet.EXPECT().GetVFLinkNames(tc.podSriovVFDeviceID).Return(sriovVfNetdeviceName, nil).Times(1)
+				fakeNetlink.EXPECT().LinkByName(sriovVfNetdeviceName).Return(vfDeviceLink, nil).Times(1)
+				moveVFtoNS = true
+			}
+			if moveVFtoNS {
+				fakeNetlink.EXPECT().LinkByName(sriovVfNetdeviceName).Return(vfDeviceLink, nil).Times(1)
+				fakeNetlink.EXPECT().LinkSetNsFd(vfDeviceLink, gomock.Any()).Return(nil).Times(1)
+				containerInterfaceLink := &netlink.Dummy{LinkAttrs: netlink.LinkAttrs{Index: 2, MTU: mtu, HardwareAddr: containerVethMac, Name: containerIfaceName, Flags: net.FlagUp}}
+				fakeNetlink.EXPECT().LinkByName(containerIfaceName).Return(containerInterfaceLink, nil).Times(1)
+				fakeNetlink.EXPECT().LinkSetMTU(containerInterfaceLink, gomock.Any()).Return(nil).Times(1)
+				fakeNetlink.EXPECT().LinkSetUp(containerInterfaceLink).Return(nil).Times(1)
+			}
+			err := testIfConfigurator.configureContainerLink(podName, testPodNamespace, podContainerID, containerNS.Path(), containerIfaceName, mtu, tc.sriovVFDeviceID, tc.podSriovVFDeviceID, ipamResult, nil)
+			if tc.expectErr != nil {
+				assert.Error(t, err)
+				assert.Equal(t, tc.expectErr, err)
+			} else {
+				assert.NoError(t, err)
+			}
+			err = containerNS.(*fakeNS).clear()
+			assert.NoError(t, err)
```

Comment:
this could be defered after the ns is created to be safe and more readable.

Reply:
updated.